ray-operator/controllers/ray/utils/validation.go

```diff
@@ -164,3 +137,47 @@ func ValidateRayServiceSpec(rayService *rayv1.RayService) error {
 	}
 	return nil
 }
+
+func ValidateGCSFaultTolerance(spec *rayv1.RayClusterSpec, annotations map[string]string) error {
```

Comment:
Put GCSFaultTolerance-related checks in a function to make it reusable in RayCluster and RayJob.